indigo_app/static/javascript/indigo/models.js

```diff
@@ -422,19 +422,19 @@
 
   /** Create a new document by parsing an frbr URI */
   Indigo.Document.newFromFrbrUri = function(frbr_uri) {
-    // /za-cpt/act/by-law/2011/foo
+    // /akn/za-cpt/act/by-law/2011/foo
     var parts = frbr_uri.split('/'),
-        tmp = parts[1].split('-'),
-        country = tmp[0],
-        locality = tmp.length > 1 ? tmp[1] : null,
-        bump = parts.length > 5 ? 1 : 0;
+        place = parts[2].split('-'),
+        country = place[0],
+        locality = place.length > 1 ? place[1] : null,
```

Comment:
If the place code is `za-foo-bar` this breaks. split() in JS isn't nearly as nice as in python, you can't tell it to split once. Easiest is probably to take the substring after the first `-`.